tutorial/tests/test_functional_programming.py

```diff
@@ -11,7 +11,7 @@
 from numpy.typing import NDArray
 
 
-def reference_solution_exercise2(my_list: List[int], k: int):
+def reference_exercise2(my_list: List[int], k: int):
```

Comment:
This definition should be moved closer to the respective test (above **line 71**). Also:

```suggestion
def reference_exercise2(my_list: List[int], k: int) -> List[int]:
```